panic: invalid conn id length

This was introduced by #3529.

```
MITM test with QUIC version v2 unsuccessful attacks corrupting packets 
  downloads a message when packet are corrupted towards the server
  /home/circleci/project/integrationtests/self/mitm_test.go:246
panic: invalid conn id length

goroutine 46470 [running]:
github.com/lucas-clemente/quic-go/internal/protocol.ParseConnectionID(...)
	/home/circleci/project/internal/protocol/connection_id.go:52
github.com/lucas-clemente/quic-go/internal/wire.ParseConnectionID({0xc004158600?, 0x22c6df02e8?, 0x0?}, 0x56579f?)
	/home/circleci/project/internal/wire/header.go:35 +0x22f
github.com/lucas-clemente/quic-go.(*packetHandlerMap).handlePacket(0xc007b67d90, 0xc004409f80)
	/home/circleci/project/packet_handler_map.go:380 +0x65
github.com/lucas-clemente/quic-go.(*packetHandlerMap).listen(0xc007b67d90)
	/home/circleci/project/packet_handler_map.go:375 +0xc5
created by github.com/lucas-clemente/quic-go.newPacketHandlerMap
	/home/circleci/project/packet_handler_map.go:140 +0x385

--- a/internal/wire/header.go
+++ b/internal/wire/header.go
@@ -29,6 +29,9 @@ func ParseConnectionID(data []byte, shortHeaderConnIDLen int) (protocol.Connecti
 		return protocol.ConnectionID{}, io.EOF
 	}
 	destConnIDLen := int(data[5])
+	if destConnIDLen > protocol.MaxConnIDLen {
+		return protocol.ConnectionID{}, protocol.ErrInvalidConnectionIDLen
+	}
 	if len(data) < 6+destConnIDLen {
 		return protocol.ConnectionID{}, io.EOF
 	}

--- a/internal/wire/header_test.go
+++ b/internal/wire/header_test.go
@@ -86,6 +86,15 @@ var _ = Describe("Header Parsing", func() {
 				Expect(err).To(MatchError(io.EOF))
 			}
 		})
+
+		It("errors when encountering a too long connection ID", func() {
+			b := []byte{0x80, 0, 0, 0, 0}
+			binary.BigEndian.PutUint32(b[1:], uint32(protocol.Version1))
+			b = append(b, 21) // dest conn id len
+			b = append(b, make([]byte, 21)...)
+			_, err := ParseConnectionID(b, 4)
+			Expect(err).To(MatchError(protocol.ErrInvalidConnectionIDLen))
+		})
 	})
 
 	Context("identifying 0-RTT packets", func() {

